codegen/generate.pl

```diff
@@ -28,17 +35,40 @@ sub gen_header {
 };
 }
 
+use constant GEN_INFO_FILE => "./codegen_hashes.tmp";
+# Compile info
+my %info = (
+    version => VERSION,
+);
+# Load info from previous compilation if it exists
+if (-e GEN_INFO_FILE) {
+    print "[prep] Loading data from previous compilation...";
+    # eval to catch it if the GEN_INFO_FILE is corrupt.
+    # Insane edge case as no one should ever edit it.
+    # But, better safe than sorry.
+    eval {
+        my $old_info = LoadFile GEN_INFO_FILE;
+        if ($old_info->{version} eq $info{version}) {
+            print "usable!\n";
+            # There is a potential for carrying over arbitrary data here.
+            # But not in any dangerous way because of the use case.
+            $info{hashes} = $old_info->{hashes};
+        } else {
+            print "not usable.\n";
+        } 1;
+    } or do {
+        print "fail.\n"
+    }
+}
+
 my $input_file_count = scalar @ARGV;
 my $i = 0;
 
 foreach my $filename (@ARGV) {
     # Terminal output non-sense.
     $i++;
     my $progress = int(($i*100)/$input_file_count);
-    printf "[%3d%s] Building CPP files for %s\n", $progress, "%", $filename;
-
-    # Load yaml definition.
-    my $file = LoadFile $filename;
+    printf "[%3d%%] Building CPP files for %s...", $progress, $filename;
 
     # Get path without file extension.
```

Comment:
Look into `File::Basename::fileparse`. Here's an example from one of my scripts: https://github.com/Mimickal/FunboxScripts/blob/master/change_container.pl#L117